myapp/main.py

```diff
@@ -0,0 +1,490 @@
+# Initialize an app
+app = Flask(__name__)
+app.debug = True
+app.secret_key = 'A0Zr98j/3yX R~XHH!jmN]LWX/,?RT'
+
+
+# 1.Commands
+# Utility method for creating all the tables in the database
+@app.route('/admin/bootstrap')
+def bootstrap():
+    schema.Base.metadata.create_all(db.engine)
+    return "Bootsrapped!"
+
+
+# Clean all the rows in Lab_rows and Lab_desc
+@app.route('/clean_database', methods=['GET'])
+def clean_database():
+    db_session = db.get_session()
+    for r in db_session.query(schema.Lab_rows):
+        db_session.delete(r)
+    for d in db_session.query(schema.Lab_desc):
+        db_session.delete(d)
+    db_session.commit()
+    return "Database has been cleaned"
+
+
+# 2.Common Side
+# common homepage(temporary)
+@app.route('/')
+def main():
+    return render_template('common_home.html')
+
+
+# 3.Student Side
+# student homepage
+@app.route('/student_home')
+def student_home():
+    return render_template('student_home.html')
+
+
+# A student can select among all current available labs
+@app.route('/student_enter_data')
+def student_enter_data():
+    lab_list = []
+    db_session = db.get_session()
+    query_all_lab = db_session.query(schema.Lab_info).all()
+    for lab in query_all_lab:
+        lab_list.append({'lab_name': lab.lab_name,
+                         'class_name': lab.class_name,
+                         'prof_name': lab.prof_name})
+    return render_template('student_enter_data.html',
+                           lab_list=lab_list)
+
+
+# A student can input data here
+@app.route('/student_data_entry/<lab_id>')
+def student_data_entry(lab_id):
+    rows_info = []
+    lab_info = []
+    db_session = db.get_session()
+    query_lab_info = db_session.query(schema.Lab_info).filter(
+        schema.Lab_info.lab_id == lab_id)
+    query_rows_info = db_session.query(schema.Lab_rows).filter(
+        schema.Lab_rows.lab_id == lab_id).order_by(schema.Lab_rows.row_order)
+
+    # Convert the objects to dictionaries to make them JSON serializable
+    for row in query_rows_info:
+        rows_info.append({'row_name': row.row_name, 'row_desc': row.row_desc,
+                          'value_type': row.value_type,
+                          'value_range': row.value_range,
+                          'value_candidates': row.value_candidates})
+    for lab in query_lab_info:
+        lab_info.append({'lab_id': lab.lab_id,
+                         'lab_name': lab.lab_name,
+                         'lab_desc': lab.lab_desc})
+    return render_template('student_data_entry.html',
+                           rows_info=rows_info, lab_info=lab_info)
+
+
+# store incoming data into the database
+@app.route('/_student_receive_data', methods=['POST'])
+def _student_receive_data():
+    jsonData = request.get_json()
+    lab_data = jsonData['lab_data']
+    db_session = db.get_session()
+    for d in lab_data:
+        student_name = d['student_name']
+        for r in d['rows_info']:
+            lab_id = r['lab_id']
```

Comment:
what happens when a user supplies a POST whose JSON body does not have a `lab_data` attribute? What if `lab_data` is not an array?